nengo/cache.py

```diff
@@ -205,49 +314,80 @@ def __delitem__(self, key):
         self._deletes.add(key)
 
     def remove_file_entry(self, filename):
+        """Remove entries mapping to ``filename``."""
         self._removed_files.add(filename)
 
     def __enter__(self):
         with self._lock:
-            self._index = self._load_index()
+            try:
+                self._load_index()
+            except:
+                self._reinit()
+                self._index = {}
         return self
 
+    def _reinit(self):
```

Comment:
Could you add a comment about what this is doing? What is it reinitializing specifically?